### DISCUSSION TITLE
Help please

### DISCUSSION THREAD

**Discussion by HackerEnvy** [sherlock-project/sherlock#929] [Q&A]
When I try to do python3 -m pip install -r requirements.txt  it says 

Defaulting to user installation because normal site-packages is not writeable
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/beautifulsoup4/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/beautifulsoup4/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/beautifulsoup4/    
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/beautifulsoup4/    
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/beautifulsoup4/    
Could not fetch URL https://pypi.org/simple/beautifulsoup4/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/beautifulsoup4/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
ERROR: Could not find a version that satisfies the requirement beautifulsoup4>=4.8.0 (from -r requirements.txt (line 1)) (from versions: none)        
ERROR: No matching distribution found for beautifulsoup4>=4.8.0 (from -r requirements.txt (line 1))                                                   
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.                                    
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping

**Comment by dotslash-web** [comment]
Explore this thread on [Stack Overflow](https://stackoverflow.com/questions/45954528/pip-is-configured-with-locations-that-require-tls-ssl-however-the-ssl-module-in)

**Comment by Cfomodz** [comment]
Sherlock no longer uses `requirements.txt`.

Install using the recommended method:

```bash
pipx install sherlock-project
```

**Comment by ayuuXploits** [comment]
The error "Can't connect to HTTPS URL because the SSL module is not available" occurs when Python lacks OpenSSL support, preventing pip from making secure HTTPS connections.

Fix based on your OS:

Ubuntu / Debian:
sudo apt update && sudo apt install -y libssl-dev openssl python3-pip

macOS (Homebrew):
brew install openssl && brew reinstall python

Windows:
Re-run the official installer from python.org, select "Modify", and make sure "pip", "tcl/tk and IDLE", and "Add Python to PATH" are checked. If broken, perform a clean reinstall.

Verify the fix by running:
python3 -c "import ssl; print(ssl.OPENSSL_VERSION)"

Once working, install the dependencies:
python3 -m pip install -r requirements.txt
